### PR TITLE
Evaluation result classes shouldn't be nested under the Evaluator cla…

### LINKED ISSUE #1280: Evaluation result classes shouldn't be nested under the Evaluator classes
When using our evaluation API:

```C#
BinaryClassificationContext binClassificationCtx = new BinaryClassificationContext(env);
BinaryClassifierEvaluator.CalibratedResult metrics = binClassificationCtx.Evaluate(predictions, "Label");
```

It is quite odd for the `metrics` object to be a nested class in the `BinaryClassifierEvaluator`. I didn't even use the `BinaryClassifierEvaluator` class at all, why am I dealing with a nested class inside of it?

I think these result classes should be moved to no longer be nested classes. Instead we should just have a `BinaryClassifierEvaluationResult` class.

/cc @Zruty0 @TomFinley @KrzysztofCwalina @terrajobst

### PR DESCRIPTION
**Pull request by anupambhatta**
…sses #1280

1. Moved nested classes Result and CalibratedResult.
2. Refactored to EvaluatorMetrics and BinaryClassificationMetrics
3. Fixes #1280 

We are excited to review your PR.

So we can do the best job, please check:

- [x] There's a descriptive title that will make sense to other developers some time from now. 
- [x] There's associated issues. All PR's should have issue(s) associated - unless a trivial self-evident change such as fixing a typo. You can use the format `Fixes #nnnn` in your description to cause GitHub to automatically close the issue(s) when your PR is merged.
- [x] Your change description explains what the change does, why you chose your approach, and anything else that reviewers should know.
- [x] You have included any necessary tests in the same PR.

### CODE REVIEW

**Comment by sfilipi** [comment]
> namespace Microsoft.ML.Data.Evaluators

I'd keep it on the same namespace as the Evaluators. 

---
Refers to: src/Microsoft.ML.Data/Evaluators/BinaryClassificationMetrics.cs:1 in d7110a5. [](commit_id = d7110a559124f10e0068d180b15dfce07b25a3ee, deletion_comment = False)

File: src/Microsoft.ML.Data/Evaluators/BinaryClassificationMetrics.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+﻿namespace Microsoft.ML.Data.Evaluators
+{
+    public sealed class BinaryClassificationMetrics : EvaluatorMetrics
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/Evaluators/BinaryClassificationMetrics.cs]: >BinaryClassificationMetrics [](start = 24, length = 27)

Might be good to preserve part of the the class name "CalibratedResult"

File: src/Microsoft.ML.Data/Evaluators/EvaluatorMetrics.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Data.Evaluators
+{
+    public class EvaluatorMetrics
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/Evaluators/EvaluatorMetrics.cs]: >EvaluatorMetrics [](start = 17, length = 16)

maybe call this one BinaryClassificationMetrics, and the other CalibratedBinaryClassificationMetrics.

**Comment by sfilipi** [comment]
Thanks for the contribution @bhattacharjeeanupam2@gmail.com 

---
In reply to: [431610622](https://github.com/dotnet/machinelearning/pull/1330#issuecomment-431610622) [](ancestors = 431610622)

**Comment by anupambhatta** [comment]
@sfilipi sure, I will make sure it builds and yes will change the names as per your suggestion.
@Zruty0 Yes will do changes to them as well. Will follow similar naming patterns as Binary.

**Comment by Ivanidzo4ka** [comment]
> namespace Microsoft.ML.Data.Evaluators

please add header to the file:
```
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.
```

---
Refers to: src/Microsoft.ML.Data/Evaluators/BinaryClassificationMetrics.cs:1 in d7110a5. [](commit_id = d7110a559124f10e0068d180b15dfce07b25a3ee, deletion_comment = False)

**Comment by Ivanidzo4ka** [comment]
> using System;

please add header to the file:
```
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.
```

---
Refers to: src/Microsoft.ML.Data/Evaluators/EvaluatorMetrics.cs:1 in d7110a5. [](commit_id = d7110a559124f10e0068d180b15dfce07b25a3ee, deletion_comment = False)

**Comment by sfilipi** [comment]
Hi @anupambhatta . Are you still planning to continue this PR?

**Comment by shauheen** [comment]
@anupambhatta please lets us know if you are planning to complete this PR.

**Pull request closed without merging**